### DISCUSSION TITLE
su2 compilation always needs coolprop library

### DISCUSSION THREAD

**Discussion by bigfooted** [su2code/SU2#1869] [General]
It looks like compilation (or actually meson.py) depends on being able to reach the coolprop library, even when it is not used:
https://www.cfd-online.com/Forums/su2-installation/246859-coolprop-link-7-5-0-doesnt-work.html

I already noticed when compiling (in linux, not windows as the cfd-online post) that I always see some coolprop message. I guess some coolprop actions need to be inside some 'if coolprop' condition, depending on the presence of the coolprop compiler option (-Denable-coolprop)

Besides that, it looks like the auto-fetch of coolprop in windows does not work, at least for the cfd-online user.

So I've had a look at meson_scripts/init.py 
So what I understand is that we always download all the M++ and coolprop packages, even though they are not necessary for compilation and running. Is it really necessary to download these packages, or what's the reason for always doing this?

**Comment by pcarruscag** [comment]
We also download codi even without AD

**Reply by bigfooted** [reply]
Yes, that's right, I was focusing on the externals folder. So what's the difference between a package residing in externals and a package in subprojects? M++ and coolprop are both in the externals folder, codi is in subprojects, like meson and ninja. So my main question is actually, do we want to download packages even if we do not need it?

**Reply by pcarruscag** [reply]
There's no difference. I would suggest not fixing what is not broken.
